---
title: "Project Viva: age 7 DNAm and metals EWAS"
output:
  html_document:
    toc: true
    toc_float: true
---


## Required Packages and functions

```{r library,warning=FALSE,message=FALSE,eval=FALSE}
# libraries
library(sas7bdat)
library(sva)
library(reshape)
library(ggplot2)
library(gridExtra)
library(knitr)
library(dplyr)
library(kableExtra)
library(tidyverse)
library(minfi)
library(stringr)
library(limma)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
library(minfi)
library(DMRcate)
library(UpSetR)
library(reshape)
library(corrplot)
library(factoextra)
library(ENmix)

# loading annotation
anno = getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
anno = data.frame(anno)

# functions
# manhattan polt
manhattan = function(probe, region, FDR = FALSE, annotate = NULL, title = NULL){
	# chromosome as numeric
	probe$chr = as.numeric(gsub("chr", "", probe$chr))
	# dataframes with common columns
	probe = data.frame(name = probe$cpg, p = probe$P.Value, fdr = probe$adj.P.Val, bonf = probe$adj.P.Val.bonf, chr = probe$chr, pos = probe$pos, type = 'position', dmp_sig = NA, color = NA)
	region_df = data.frame(name = as.character(seq(1:nrow(region))), p = NA, fdr = NA, bonf = NA, chr = region$chr, pos = region$start, type = 'region', dmp_sig = NA, color = NA, size = NA)
	# indicating probes within regions
	for (i in 1:length(region$chr)){
		probe$dmp_sig[(probe$chr == region$chr[i]) & (probe$pos >= region$start[i]) & (probe$pos <= region$end[i])] = 1
	}
	# variable for point color
	probe$color[probe$dmp_sig == 1] = 50
	probe$color[probe$fdr < 0.05] = 100
	probe$color[is.na(probe$color)] = probe$chr[is.na(probe$color)]
	# variable for point size
	probe$size[probe$color == 50 | probe$color == 100] = 1
	probe$size[is.na(probe$size)] = 0.5
	# combine dataframes
	df_comb = rbind(probe, region_df)
	# dataset for plotting
		don = df_comb %>% 
	  		# Compute chromosome size
				group_by(chr) %>% summarise(chr_len=as.numeric(max(pos))) %>% 
	  		# Calculate cumulative position of each chromosome
	 			mutate(tot=cumsum(chr_len)-chr_len) %>% dplyr::select(-chr_len) %>%
	  		# Add this info to the initial dataset
	  			left_join(df_comb, ., by=c("chr"="chr")) %>%
	  		# Add a cumulative position of each site
				arrange(chr, pos) %>% mutate(poscum=pos+tot) # %>%
	 		# Prepare X axis
				axisdf = don %>% group_by(chr) %>% summarize(center=(max(poscum) + min(poscum))/2)
		don = merge(don, df_comb, on='name', all.x=T)
		don = don[order(don$size),]
		don_position = don[don$type == 'position',]
		don_region = don[don$type == 'region',]
	
		colors = c("#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", "#969696", "#737373", '#2166ac', 'black')
	
	manhattan = ggplot(don_position, aes(x=poscum, y=-log10(p))) +
	geom_point(aes(color=as.factor(color)), size= don_position$size, alpha = don_position$size) + scale_color_manual(values = colors) +
    # p-value cutoffs
	geom_hline(yintercept=-log10(0.05/nrow(don_position)), colour = '#AB3428', size=.2, alpha = 0.5) +
	geom_vline(xintercept= don_region$poscum, colour = '#4393c3', size=.2) +
	# custom axes:
	scale_x_continuous(expand = c(0.005, 0.005), limits = c(min(don_position$poscum), max(don_position$poscum)), label = axisdf$chr, breaks= axisdf$center) +
	scale_y_continuous(expand = c(0, 0), limits = c(0, (max(-log10(don_position$p)) + 0.5)), breaks = seq(from = 0, to = (max(-log10(don_position$p)) + 0.5), by = 1)) +
	# Custom theme:
    theme_minimal() + theme( 
	legend.position="none", panel.border = element_blank(), panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(), panel.grid.major.y = element_blank(), text = element_text(size = 7.5)) + 
    labs(y=expression(-log[10](italic(p))), x='Chromosome') 
    if (!is.null(title)){
    	manhattan = manhattan + labs(title = title)
    }
    if (FDR == TRUE){
    	manhattan = manhattan + geom_hline(yintercept=-log10(max(don_position$p[don_position$fdr < 0.05])), colour='#AB3428', size=.2, alpha = 0.5, linetype = "dashed")
    } 
    if (!is.null(annotate)){
		manhattan = manhattan + annotate("text", x = max(don_position$poscum)*0.05, y = max(-log10(don_position$p)), label = annotate, size = 4)
	}
	return(manhattan)
}
# lambda
lambda = function(p) median(qchisq(p, df=1, lower.tail=FALSE), na.rm=TRUE) / qchisq(0.5, df=1)
gg_qqplot = function(pvector){
	l = round(lambda(pvector), 3)
	o = -log10(sort(pvector, decreasing = FALSE))
	e = -log10(ppoints(length(pvector)))
	df = data.frame(o = o, e = e)
	ggplot(df, aes(e, o)) + geom_point(alpha = 0.5, size = 1) + geom_abline(intercept = 0, slope = 1, color = '#AB3428') + labs(y = expression(Observed ~ ~-log[10](italic(p))), x = expression(Expected ~ ~-log[10](italic(p)))) + theme_classic() + annotate("text", x = 1, y = 5, label = paste0('lambda = ', l))
}
# volcano
volcano = function(probe, type = 'DMP'){
	if (type == 'DMP'){
		volcano = ggplot(probe, aes(x=logFC, y  = -log10(P.Value))) + 
		geom_point(size = 0.8, alpha=0.4) + 
		geom_hline(aes(yintercept = -log10(0.05/nrow(probe)),linetype = 'Bonferroni threshold'), color = "#AB3428", size = 0.5) + 
	#	geom_hline(aes(yintercept = -log10(max(P.Value[adj.P.Val < 0.05])),linetype = 'FDR threshold'), color = "#AB3428", size = 0.3) + 
		scale_linetype_manual(name = '', values = c(1,2), guide = guide_legend(override.aes = list(color = c("#AB3428", "#AB3428")))) + theme_minimal() + 
		labs(y=expression(-log[10]*"(P-value)"), x='Effect estimate') + theme(panel.grid.minor.y = element_blank()) + theme(text = element_text(size=8)) + 
		theme(legend.position="none") + scale_y_continuous(expand = c(0, 0)) + theme(panel.grid.minor.x = element_blank(), panel.grid.major.y = element_line(size = 0.2, color = 'gray65'), panel.grid.major.x = element_line(size = 0.2, color = 'gray65'))
		return(volcano)
	} else if (type == 'DVP'){
		volcano = ggplot(probe, aes(x= DiffLevene, y  = -log10(P.Value))) + 
		geom_point(size = 0.8, alpha=0.4) + 
		geom_hline(aes(yintercept = -log10(0.05/nrow(probe)),linetype = 'Bonferroni threshold'), color = "#AB3428", size = 0.5) + 
		geom_hline(aes(yintercept = -log10(max(P.Value[Adj.P.Value < 0.05])),linetype = 'FDR threshold'), color = "#AB3428", size = 0.3) + 
		scale_linetype_manual(name = '', values = c(1,2), guide = guide_legend(override.aes = list(color = c("#AB3428", "#AB3428")))) + theme_minimal() + 
		labs(y=expression(-log[10]*"(P-value)"), x='Effect estimate') + theme(panel.grid.minor.y = element_blank()) + theme(text = element_text(size=8)) + 
		theme(legend.position="none") + scale_y_continuous(expand = c(0, 0)) + theme(panel.grid.minor.x = element_blank(), panel.grid.major.y = element_line(size = 0.2, color = 'gray65'), panel.grid.major.x = element_line(size = 0.2, color = 'gray65'))
		return(volcano)
	}
}
# DMP analysis
run_DMP <- function(mvals, design){
  # fit model
  l_fit <- limma::lmFit(object = mvals, design = design)
  
  # extract standard errors
  std_err <- l_fit$stdev.unscaled[,2]*l_fit$sigma
  std_err_df <- data.frame(std_err)
  std_err_df$cpg <- rownames(std_err_df)
  
  e_fit <- limma::eBayes(l_fit, robust = TRUE)
  
  # extract results and add Bonferroni correction
  p_top <- limma::topTable(e_fit, adjust = "BH", coef = 2, num = Inf, confint = TRUE)
  p_top <- p_top[order(p_top$P.Value), , drop = FALSE]
  p_top$adj.P.Val.bonf <- topTable(e_fit, adjust="bonferroni", coef=2, number = Inf)$adj.P.Val
  
  # merge results and standard errors
  p_top$cpg <- rownames(p_top)
  p_top <- merge(p_top, std_err_df, by = 'cpg')
  rownames(p_top) <- p_top$cpg
  
  return(p_top)
}
# Combp
acf.table<-function(x,loc,dist.cutoff){
  flag=TRUE; lag=1; result=NULL
  while(flag){
    x1=head(x,-lag); x2=tail(x,-lag); dist=diff(loc,lag=lag)
    index=(dist<dist.cutoff)  
    if(all(!index)){flag=FALSE}else{
      result=rbind(result,data.frame(x1=x1[index],x2=x2[index],dist=dist[index]))
    lag=lag+1
    }
  }
  return(result)  
}
get.acf<-function(data,dist.cutoff,bin.size){
  temp<-NULL
  for (chr in unique(as.vector(data$chr))){
    y<-data[as.vector(data$chr)==chr,]; y<-y[order(y$end),]
    temp<-rbind(temp,acf.table(y$p,y$end,dist.cutoff))
  }
  bin.label<-findInterval(temp$dist,seq(bin.size,dist.cutoff,bin.size))
  temp.stouffer<-by(temp,bin.label,FUN=function(x){cor.test(qnorm(x$x1),
               qnorm(x$x2),alternative="greater")},simplify=FALSE)
  cor.stouffer<-sapply(temp.stouffer,function(x){x$estimate})
  p.stouffer<-sapply(temp.stouffer,function(x){x$p.value})
  if (any(p.stouffer>0.05)){
    index=min(which(p.stouffer>0.05))
    cor.stouffer[index:length(cor.stouffer)]=0
  }
  return(cor.stouffer)
}
regplot<-function(ref,sig,extend=2000,outf="region_plot.pdf"){
  sig=sig[order(sig[,"chr"],sig[,"start"]),]
  ref=ref[order(ref[,"chr"],ref[,"start"]),]
  pdf(outf)
  for(i in 1:nrow(sig)){
    chr=as.vector(sig$chr[i])
    pos1=sig$start[i]
    pos2=sig$end[i]
    subset=ref[as.vector(ref$chr)==chr & ref$start>=(pos1-extend) & ref$start<=(pos2+extend),]
    subset$cor="black"
    subset$cor[subset$start>=pos1 &subset$start<=pos2]="red"
    ylab=bquote('-log'['10']*'(P) value')
    plot(subset$start,-log10(subset$p),col=subset$cor,pch=20,xlim=c(pos1-extend,
          pos2+extend),xlab="Chromosome position",ylab=ylab)
  }
  dev.off()
}
# comb_p-like method
combp2<-function(data,dist.cutoff=1000,bin.size=310,seed=0.01,
               region_plot=TRUE,mht_plot=TRUE,nCores=10,verbose=TRUE){
  if(nCores>detectCores()){nCores=detectCores()}
  data=as.data.frame(data)
  data$start=as.numeric(as.vector(data$start))
  data$end=as.numeric(as.vector(data$end))
  data=data[!is.na(data$start) & !is.na(data$end),]
  data$p=as.numeric(as.vector(data$p))
  acf<-get.acf(data,dist.cutoff,bin.size)
  if(verbose){
    cat("P value correlations:\n")
    bin=seq(bin.size,dist.cutoff,bin.size)
    if(!(dist.cutoff%%bin.size==0)){bin=c(bin,dist.cutoff)}
    print(data.frame(bin=bin,acf=acf))
  }
  result<-mclapply(unique(as.vector(data$chr)), function(chr){
    y=data[as.vector(data$chr)==chr,]; y=y[order(y$end),]
    pos=y$end; p=qnorm(y$p)
    temp=sapply(pos,function(i){
      index.i=(abs(pos-i)<bin.size);
      if (sum(index.i)>1){  
        int<-findInterval(c(dist(pos[index.i])),c(bin.size,2*bin.size))
        sd<-sqrt(sum(acf[int+1])*2+sum(index.i))
        return(pnorm(sum(p[index.i]),mean=0,sd=sd))
      }else{return(y$p[index.i])}
    })
    return(data.frame(chr,start=pos,end=pos,s.p=temp))
  },mc.cores=nCores)
  result <- do.call("rbind", result)
  names(result)=c("chr","start","end","s.p")
  result=result[p.adjust(result$s.p,method="fdr")<seed,]
  result.fdr=NULL
  if (nrow(result)>0){
    for (chr in unique(result$"chr")){
      y=data[as.vector(data$chr)==chr,]; y=y[order(y$end),]
      pos=y$end; p=qnorm(y$p)
      result.chr=result[result$"chr"==chr,]
      a=IRanges::IRanges(start=result.chr$start,end=result.chr$end)
      b=IRanges::reduce(a,min.gapwidth=dist.cutoff)
      start=IRanges::start(b); end=IRanges::end(b)
      region.max<-max(IRanges::width(b))
      temp=sapply(1:length(b),function(i){
        index.i=(pos>=start[i] & pos<=end[i]);
        if (sum(index.i)>1){  
          int<-findInterval(c(dist(pos[index.i])),
              seq(bin.size,region.max+bin.size,bin.size))
          sd<-sqrt(sum(ifelse(int<length(acf),
              acf[int+1],0))*2+sum(index.i))
          return(pnorm(sum(p[index.i]),mean=0,sd=sd))
        }else{return(y$p[index.i])}
      })
      result.fdr=rbind(result.fdr,data.frame(chr,start,end,p=temp))
    }
    
    result.fdr$length = (result.fdr$end - result.fdr$start) + 1
    result.fdr = result.fdr[result.fdr$length > 1,]
    ##### BH FDR correction and Sidak correction
    result.fdr$fdr=p.adjust(result.fdr$p,method="fdr")
    result.fdr$sidak=(1-(1-result.fdr$p)^(nrow(data)/(result.fdr$end-result.fdr$start+1)))
    result.fdr<-result.fdr[order(result.fdr$p),]
    ##### use 0-coordinate
    result.fdr$start=(result.fdr$start-1)
  }
  if(is.null(result.fdr)){cat("Number of identified DMR:  0\n")}else{
    ndmr=nrow(result.fdr)
  result.fdr$start=as.numeric(as.vector(result.fdr$start))
  result.fdr$end=as.numeric(as.vector(result.fdr$end))
  result.fdr$chr=factor(result.fdr$chr)
    cat("Number of DMRs identified:  ",ndmr, "\n")
    if(region_plot){
      cat("Drawing regional plot: region_plot.pdf ...\n")
      sig=result.fdr
      regplot(ref=data,sig)
    }
  if(mht_plot){
    cat("Drawing manhattan plot: mht.jpg ...\n")
    set2=NULL
    for(i in 1:ndmr){
        set2=c(set2,as.vector(data$probe[as.vector(data$chr)==as.vector(result.fdr$chr[i])
           & data$start>=result.fdr$start[i] & data$start<=result.fdr$end[i]]))
    }
  mhtplot(probe=data$probe,chr=as.vector(data$chr),pos=data$start,p=data$p,color="gray",markprobe=set2)
  }
  #number of probes within eath DMR
  result.fdr$nprobe=NA
  for(i in 1:nrow(result.fdr)){
result.fdr$nprobe[i]=nrow(data[as.vector(data$chr)==as.vector(result.fdr$chr[i])
& data$start>=result.fdr$start[i] & data$end<=result.fdr$end[i],])
}
  write.table(result.fdr,"resu_combp.csv",row.names=FALSE,sep=",")
  }
}
```


## Load DNAm and pheno data

```{r data,warning=FALSE,message=FALSE,eval= FALSE}
load("/Users/annebozack/Box/NIEHS-R01 ONES/Methylation Data/Age7Blood_ComBat_Betas_Mvlas_filteredPorbes_metalAnalaysis.RData")

dim(pDatAge7Metal)
# 333 164

dim(ComBat.Mvalues.Metals)
# 394460    333

rownames(pDatAge7Metal) = pDatAge7Metal$samplename
```


## EWAS

### Mn

```{r, warning=FALSE,message=FALSE,eval=FALSE}
# Unadjusted
modUnadj = model.matrix(~ pDatAge7Metal$Mn_log2)

DMP_Mn_unadj <- run_DMP(mvals = ComBat.Mvalues.Metals, design = modUnadj)

table(DMP_Mn_unadj$P.Value < 0.05)
 # FALSE   TRUE 
# 366477  27983 

table(DMP_Mn_unadj$adj.P.Val < 0.05)
 # FALSE   TRUE 
# 394454      6 

# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdj = model.matrix(~ Mn_log2 + female_d + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7Metal)

dim(modAdj)
# 333  20

# M-values for comoplete cases
ComBat.Mvalues.noMissAdj = ComBat.Mvalues.Metals[,colnames(ComBat.Mvalues.Metals) %in% rownames(modAdj)]
ComBat.Mvalues.noMissAdj <- ComBat.Mvalues.noMissAdj[,match(rownames(modAdj), colnames(ComBat.Mvalues.noMissAdj))]
all(rownames(modAdj)==colnames(ComBat.Mvalues.noMissAdj))
# TRUE 
identical(rownames(modAdj),colnames(ComBat.Mvalues.noMissAdj))
# TRUE 

DMP_Mn_Adj <- run_DMP(mvals = ComBat.Mvalues.noMissAdj, design = modAdj)

table(DMP_Mn_Adj$P.Value < 0.05)
#  FALSE   TRUE 
# 354938  39522 

table(DMP_Mn_Adj$adj.P.Val < 0.05)
#   FALSE   TRUE 
# 394447     13 

DMP_Mn_Adj_sig = DMP_Mn_Adj[DMP_Mn_Adj$adj.P.Val < 0.05,]
DMP_Mn_Adj_sig = cbind(DMP_Mn_Adj_sig, anno[,'chr'][match(rownames(DMP_Mn_Adj_sig), rownames(anno))])
DMP_Mn_Adj_sig = cbind(DMP_Mn_Adj_sig, anno[,'pos'][match(rownames(DMP_Mn_Adj_sig), rownames(anno))])
DMP_Mn_Adj_sig = cbind(DMP_Mn_Adj_sig, anno[,'UCSC_RefGene_Name'][match(rownames(DMP_Mn_Adj_sig), rownames(anno))])
colnames(DMP_Mn_Adj_sig)[c(12,13,14)] = c('chr', 'pos', 'gene')
DMP_Mn_Adj_sig$group = 'all'

                # logFC       CI.L        CI.R    AveExpr      P.Value    adj.P.Val adj.P.Val.bonf   chr       pos                               gene
# cg00808511  0.1790764  0.1069774  0.25117536  2.8599969 1.632835e-06 4.954523e-02   6.440879e-01 chr16  52495237                          TOX3;TOX3
# cg03716590  0.1817816  0.1106850  0.25287826  5.0904967 8.220465e-07 3.242645e-02   3.242645e-01  chr7  95435309            DYNC1I1;DYNC1I1;DYNC1I1
# cg05613017  0.1990897  0.1226676  0.27551185  5.3286885 5.177595e-07 2.269282e-02   2.042354e-01 chr12  65589446                        LEMD3;LEMD3
# cg09856068 -0.1139538 -0.1575510 -0.07035670 -4.3655995 4.759858e-07 2.269282e-02   1.877574e-01  chr5 170878246                              FGF18
# cg11063328 -0.1613562 -0.2226900 -0.10002240 -6.0173655 4.039307e-07 2.269282e-02   1.593345e-01 chr11  18343203       GTF2H1;HPS5;HPS5;HPS5;GTF2H1
# cg13834844 -0.1018182 -0.1390236 -0.06461286 -5.1364932 1.420476e-07 1.400802e-02   5.603209e-02 chr10 102415320                                   
# cg14252211  0.1932401  0.1278047  0.25867558  5.8503162 1.522677e-08 3.003177e-03   6.006353e-03  chr1  25919394                                   
# cg19407717  0.3100622  0.2201358  0.39998850  4.0277865 5.777466e-11 2.278979e-05   2.278979e-05  chr2   1544120                    TPO;TPO;TPO;TPO
# cg19425870 -0.1474701 -0.2056736 -0.08926659 -6.4570149 1.023219e-06 3.669264e-02   4.036191e-01 chr11   3876808                              STIM1
# cg22586726  0.2694306  0.1612746  0.37758663 -0.9735686 1.524688e-06 4.954523e-02   6.014286e-01 chr10 134150542 LRRC27;LRRC27;LRRC27;LRRC27;LRRC27
# cg24196240  0.3278329  0.2099472  0.44571864  5.2479144 9.096707e-08 1.196096e-02   3.588287e-02 chr21  36096009                         NCRNA00160
# cg25101184  0.2483310  0.1535821  0.34308003  3.3810389 4.444305e-07 2.269282e-02   1.753100e-01 chr14  56231604                           RPL13AP3

# comparison to cord blood DMP
DMP_Mn_Adj[DMP_Mn_Adj$cpg == 'cg02042823',]
                  # cpg     logFC       CI.L      CI.R  AveExpr        t    P.Value adj.P.Val         B adj.P.Val.bonf    std_err
# cg02042823 cg02042823 0.1823146 0.04623718 0.3183919 6.127512 2.636034 0.00880242 0.3269381 -2.785646              1 0.06943475

gg_qqplot(DMP_Mn_Adj$P.Value)
quartz.save('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/QQ_DMP_Mn_adj_age7_020821.png', type = "png", dpi = 300)

volcano(DMP_Mn_Adj)
quartz.save('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/volcano_DMP_Mn_adj_age7_020821.png', type = "png", dpi = 300)

# combp
DMP_Mn_Adj = cbind(DMP_Mn_Adj, anno[,'chr'][match(rownames(DMP_Mn_Adj), rownames(anno))])
DMP_Mn_Adj = cbind(DMP_Mn_Adj, anno[,'pos'][match(rownames(DMP_Mn_Adj), rownames(anno))])
DMP_Mn_Adj$end = DMP_Mn_Adj[,13]
DMP_Mn_Adj_p = DMP_Mn_Adj[,c(12,13,14,7,1)]
colnames(DMP_Mn_Adj_p) = c( "chr","start", "end","p", "probe")
DMP_Mn_Adj_p$chr = as.numeric(gsub("chr", "", DMP_Mn_Adj_p$chr))
dir.create('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_age7')
setwd('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_age7')
combp2(DMP_Mn_Adj_p, region_plot = F, mht_plot = F, seed = 0.001, verbose = T)

# Number of DMRs identified:   8

# chr	start	end	p	length	fdr	sidak	nprobe
# 10	134150488	134150760	6.17E-12	272	4.94E-11	8.96E-09	7
# 19	21657528	21657756	6.20E-11	228	2.48E-10	1.07E-07	4
# 16	30572738	30573013	1.00E-09	275	2.68E-09	1.44E-06	5
# 10	45406680	45406847	2.46E-08	167	4.91E-08	5.80E-05	3
# 16	73100425	73100510	3.15E-08	85	5.04E-08	0.000146227	2
# 20	61659979	61660250	5.87E-08	271	7.82E-08	8.54E-05	2
# 10	35484767	35484824	0.000192214	57	0.000219673	0.735606002	3
# 15	91473290	91473365	0.000549729	75	0.000549729	0.944538975	2

# manhattan plot
region = read.csv('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_age7/resu_combp.csv')
probe = DMP_Mn_Adj
colnames(probe)[c(12,13)] = c('chr', 'pos')
quartz(width = 5.5, height = 3)
manhattan(probe, region, FDR = T)
quartz.save('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/manhattan_DMP_Mn_adj_age7_020821.png', type = "png", dpi = 300)

write.csv(DMP_Mn_Adj, '/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/EWAS results/DMP_Mn_Adj_age7_020821.csv')
rm(DMP_Mn_Adj_p, DMP_Mn_Adj);gc()


# EWAS using Beta values
# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdj = model.matrix(~ Mn_log2 + female_d + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7Metal)

dim(modAdj)
# 333  20

# Beta-values for comoplete cases
ComBat.Betas.noMissAdj = ComBat.Betas.Metals[,colnames(ComBat.Betas.Metals) %in% rownames(modAdj)]
ComBat.Betas.noMissAdj <- ComBat.Betas.noMissAdj[,match(rownames(modAdj), colnames(ComBat.Betas.noMissAdj))]
all(rownames(modAdj)==colnames(ComBat.Betas.noMissAdj))
# TRUE 
identical(rownames(modAdj),colnames(ComBat.Betas.noMissAdj))
# TRUE 

DMP_Mn_Adj_betas <- run_DMP(mvals = ComBat.Betas.noMissAdj, design = modAdj)

table(DMP_Mn_Adj_betas$P.Value < 0.05)
# FALSE   TRUE 
# 350362  44098 

table(DMP_Mn_Adj_betas$adj.P.Val < 0.05)
 # FALSE   TRUE 
# 387967   6493 

DMP_Mn_Adj_betas[DMP_Mn_Adj_betas$cpg == 'cg02042823',]
                  # cpg       logFC        CI.L       CI.R   AveExpr       t    P.Value adj.P.Val         B adj.P.Val.bonf     std_err
# cg02042823 cg02042823 0.007473515 0.001425951 0.01352108 0.9806583 2.43148 0.01559668 0.2561631 -6.996464              1 0.003077495

rm(DMP_Mn_Adj_betas, ComBat.Betas.noMissAdj);gc()

```

```{r, out.width = '60%'}
knitr::include_graphics("/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/QQ_DMP_Mn_adj_age7_020821.png")
```

```{r, out.width = '60%'}
knitr::include_graphics("/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/volcano_DMP_Mn_adj_age7_020821.png")
```

```{r, out.width = '100%'}
knitr::include_graphics("/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/markdownplots/manhattan_DMP_Mn_adj_age7_020821.png")
```

#### Mn, female

```{r, warning=FALSE,message=FALSE,eval=FALSE}

pDatAge7MetalF = pDatAge7Metal[pDatAge7Metal$female_d == 1,]

# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdjF = model.matrix(~ Mn_log2 + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7MetalF)

dim(modAdjF)
# 158  19

# M-values for complete cases
ComBat.Mvalues.noMissAdjF = ComBat.Mvalues.Metals[,colnames(ComBat.Mvalues.Metals) %in% rownames(modAdjF)]
ComBat.Mvalues.noMissAdjF <- ComBat.Mvalues.noMissAdjF[,match(rownames(modAdjF), colnames(ComBat.Mvalues.noMissAdjF))]
all(rownames(modAdjF)==colnames(ComBat.Mvalues.noMissAdjF))
# TRUE 
identical(rownames(modAdjF),colnames(ComBat.Mvalues.noMissAdjF))
# TRUE 

DMP_Mn_AdjF <- run_DMP(mvals = ComBat.Mvalues.noMissAdjF, design = modAdjF)

table(DMP_Mn_AdjF$P.Value < 0.05)
#  FALSE   TRUE 
# 355206  39254 

table(DMP_Mn_AdjF$adj.P.Val < 0.05)
 #  FALSE   TRUE 
# 394457      3 

DMP_Mn_AdjF_sig = DMP_Mn_AdjF[DMP_Mn_AdjF$adj.P.Val < 0.05,]
DMP_Mn_AdjF_sig = cbind(DMP_Mn_AdjF_sig, anno[,'chr'][match(rownames(DMP_Mn_AdjF_sig), rownames(anno))])
DMP_Mn_AdjF_sig = cbind(DMP_Mn_AdjF_sig, anno[,'pos'][match(rownames(DMP_Mn_AdjF_sig), rownames(anno))])
DMP_Mn_AdjF_sig = cbind(DMP_Mn_AdjF_sig, anno[,'UCSC_RefGene_Name'][match(rownames(DMP_Mn_AdjF_sig), rownames(anno))])
colnames(DMP_Mn_AdjF_sig)[c(12,13,14)] = c('chr', 'pos', 'gene')
DMP_Mn_AdjF_sig$group = 'female'

               # logFC       CI.L        CI.R   AveExpr      P.Value    adj.P.Val adj.P.Val.bonf   chr      pos                         gene
# cg11063328 -0.249298 -0.3264013 -0.17219466 -6.003312 2.210659e-09 0.0008720164   0.0008720164 chr11 18343203 GTF2H1;HPS5;HPS5;HPS5;GTF2H1
# cg19425870 -0.237274 -0.3130307 -0.16151741 -6.446281 6.056936e-09 0.0011946094   0.0023892188 chr11  3876808                        STIM1
# cg22396353 -0.146545 -0.1990485 -0.09404158 -6.101068 1.578093e-07 0.0207498194   0.0622494582 chr22 46692687          GTSE1;GTSE1;CN5H6.4

# comparison to cord blood DMPs
DMP_Mn_AdjF[DMP_Mn_AdjF$cpg %in% c('cg00954161', 'cg01744822', 'cg08904630', 'cg11161853', 'cg15712310', 'cg19908812', 'cg22799518', 'cg23903787', 'cg26462130'),]

                  # cpg        logFC        CI.L        CI.R    AveExpr          t      P.Value adj.P.Val         B adj.P.Val.bonf    std_err
# cg00954161 cg00954161  0.257499506  0.10305035  0.41194866  5.8701101  3.2957874 1.239920e-03 0.2461873 -1.019420              1 0.07879912
# cg01744822 cg01744822 -0.211712019 -0.31152730 -0.11189674 -3.3852542 -4.1929224 4.824120e-05 0.1992618  1.750167              1 0.05080262
# cg08904630 cg08904630  0.150103187  0.08427024  0.21593613  5.4765411  4.5072170 1.361571e-05 0.1992618  2.841780              1 0.03332822
# cg11161853 cg11161853  0.006649857 -0.03131855  0.04461826 -5.3278564  0.3462168 7.296913e-01 0.9536487 -5.737906              1 0.01879637
# cg15712310 cg15712310 -0.069144479 -0.12512890 -0.01316006 -0.8216403 -2.4414540 1.585585e-02 0.3709062 -3.125661              1 0.02822476
# cg19908812 cg19908812 -0.031378191 -0.15247060  0.08971422 -5.0154091 -0.5122469 6.092747e-01 0.9251463 -5.672775              1 0.06171388
# cg22799518 cg22799518 -0.222928095 -0.36857749 -0.07727870  6.2712681 -3.0256902 2.946157e-03 0.2644488 -1.744636              1 0.07429358
# cg23903787 cg23903787  0.141564362  0.02771074  0.25541798  2.9991042  2.4579634 1.517748e-02 0.3671148 -3.090393              1 0.05800327
# cg26462130 cg26462130  0.181427649  0.05426623  0.30858906  6.1675112  2.8204373 5.483197e-03 0.2928612 -2.259703              1 0.06482387

lambda(DMP_Mn_AdjF$P.Value)
# 1.350311

# combp
DMP_Mn_AdjF = cbind(DMP_Mn_AdjF, anno[,'chr'][match(rownames(DMP_Mn_AdjF), rownames(anno))])
DMP_Mn_AdjF = cbind(DMP_Mn_AdjF, anno[,'pos'][match(rownames(DMP_Mn_AdjF), rownames(anno))])
DMP_Mn_AdjF$end = DMP_Mn_AdjF[,13]
DMP_Mn_AdjF_p = DMP_Mn_AdjF[,c(12,13,14,7,1)]
colnames(DMP_Mn_AdjF_p) = c( "chr","start", "end","p", "probe")
DMP_Mn_AdjF_p$chr = as.numeric(gsub("chr", "", DMP_Mn_AdjF_p$chr))
dir.create('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_F_age7')
setwd('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_F_age7')

combp2(DMP_Mn_AdjF_p, region_plot = F, mht_plot = F, seed = 0.001, verbose = T)

# Number of DMRs identified:   3

# chr	start	end	p	length	fdr	sidak	nprobe
# 16	73100425	73100510	3.61E-08	85	1.08E-07	0.000167289	2
# 6	3849410	3849690	4.55E-07	280	6.83E-07	0.000641116	10
# 11	76839190	76839217	4.63E-06	27	4.63E-06	0.065420896	3

rm(DMP_Mn_AdjF_p, DMP_Mn_AdjF);gc()


# EWAS using Beta values
# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdjF = model.matrix(~ Mn_log2 + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7MetalF)

# Beta-values for comoplete cases
ComBat.Betas.noMissAdjF = ComBat.Betas.Metals[,colnames(ComBat.Betas.Metals) %in% rownames(modAdjF)]
ComBat.Betas.noMissAdjF <- ComBat.Betas.noMissAdjF[,match(rownames(modAdjF), colnames(ComBat.Betas.noMissAdjF))]
all(rownames(modAdjF)==colnames(ComBat.Betas.noMissAdjF))
# TRUE 
identical(rownames(modAdjF),colnames(ComBat.Betas.noMissAdjF))
# TRUE 

DMP_Mn_Adj_betasF <- run_DMP(mvals = ComBat.Betas.noMissAdjF, design = modAdjF)

table(DMP_Mn_Adj_betasF$P.Value < 0.05)
#  FALSE   TRUE 
# 351500  42960 

table(DMP_Mn_Adj_betasF$adj.P.Val < 0.05)
 # FALSE   TRUE 
# 387155   7305 

DMP_Mn_Adj_betasF[DMP_Mn_Adj_betasF$cpg %in% c('cg00954161', 'cg01744822', 'cg08904630', 'cg11161853', 'cg15712310', 'cg19908812', 'cg22799518', 'cg23903787', 'cg26462130'),]

                  # cpg         logFC          CI.L          CI.R    AveExpr          t      P.Value    adj.P.Val         B adj.P.Val.bonf      std_err
# cg00954161 cg00954161  0.0133401378  0.0069717474  0.0197085281 0.97858713  4.1414675 5.941182e-05 1.084015e-02 -1.502914   1.0000000000 0.0032301149
# cg01744822 cg01744822 -0.0170217995 -0.0232865443 -0.0107570547 0.09250324 -5.3718726 3.172004e-07 1.039692e-03  3.560674   0.1251228852 0.0031775327
# cg08904630 cg08904630  0.0047144826  0.0033211618  0.0061078034 0.97724118  6.6896343 4.954960e-10 1.503487e-05  9.903218   0.0001954533 0.0007051848
# cg11161853 cg11161853  0.0000966045 -0.0005510166  0.0007442256 0.02447637  0.2949143 7.684963e-01 9.736066e-01 -9.605891   1.0000000000 0.0003243811
# cg15712310 cg15712310 -0.0111615165 -0.0200069000 -0.0023161331 0.36261336 -2.4947680 1.376797e-02 2.426790e-01 -6.583367   1.0000000000 0.0044867176
# cg19908812 cg19908812 -0.0002030680 -0.0032362692  0.0028301331 0.03262774 -0.1323622 8.948881e-01 9.889625e-01 -9.640852   1.0000000000 0.0015378757
# cg22799518 cg22799518 -0.0021606037 -0.0044241107  0.0001029033 0.98520267 -1.8871918 6.120710e-02 5.014978e-01 -7.878724   1.0000000000 0.0011472636
# cg23903787 cg23903787  0.0197503033  0.0073678958  0.0321327108 0.88198757  3.1534961 1.974931e-03 7.869003e-02 -4.812126   1.0000000000 0.0062810129
# cg26462130 cg26462130  0.0097682941  0.0067956617  0.0127409265 0.98414252  6.4968246 1.338355e-09 3.519516e-05  8.924072   0.0005279275 0.0015071354

rm(DMP_Mn_Adj_betasF, ComBat.Betas.noMissAdjF);gc()

```

#### Mn, male

```{r, warning=FALSE,message=FALSE,eval=FALSE}

pDatAge7MetalM = pDatAge7Metal[pDatAge7Metal$female_d == 0,]

# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdjM = model.matrix(~ Mn_log2 + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7MetalM)

dim(modAdjM)
# 175  19

# M-values for complete cases
ComBat.Mvalues.noMissAdjM = ComBat.Mvalues.Metals[,colnames(ComBat.Mvalues.Metals) %in% rownames(modAdjM)]
ComBat.Mvalues.noMissAdjM <- ComBat.Mvalues.noMissAdjM[,match(rownames(modAdjM), colnames(ComBat.Mvalues.noMissAdjM))]
all(rownames(modAdjM)==colnames(ComBat.Mvalues.noMissAdjM))
# TRUE 
identical(rownames(modAdjM),colnames(ComBat.Mvalues.noMissAdjM))
# TRUE 

DMP_Mn_AdjM <- run_DMP(mvals = ComBat.Mvalues.noMissAdjM, design = modAdjM)

table(DMP_Mn_AdjM$P.Value < 0.05)
# FALSE   TRUE 
# 375128  19332 

table(DMP_Mn_AdjM$adj.P.Val < 0.05)
 #  FALSE  TRUE 
# 394428     32 
 
DMP_Mn_AdjM_sig = DMP_Mn_AdjM[DMP_Mn_AdjM$adj.P.Val < 0.05,]
DMP_Mn_AdjM_sig = cbind(DMP_Mn_AdjM_sig, anno[,'chr'][match(rownames(DMP_Mn_AdjM_sig), rownames(anno))])
DMP_Mn_AdjM_sig = cbind(DMP_Mn_AdjM_sig, anno[,'pos'][match(rownames(DMP_Mn_AdjM_sig), rownames(anno))])
DMP_Mn_AdjM_sig = cbind(DMP_Mn_AdjM_sig, anno[,'UCSC_RefGene_Name'][match(rownames(DMP_Mn_AdjM_sig), rownames(anno))])
colnames(DMP_Mn_AdjM_sig)[c(12,13,14)] = c('chr', 'pos', 'gene')
DMP_Mn_AdjM_sig$group = 'male'

                 # logFC       CI.L        CI.R   AveExpr      P.Value    adj.P.Val adj.P.Val.bonf   chr       pos
# cg03425412 -0.22620388 -0.3126879 -0.13971983 -5.940582 7.093419e-07 1.332414e-02   2.798070e-01 chr20  21500337
# cg04143120  0.41643112  0.2791269  0.55373536  6.792018 1.356169e-08 7.642209e-04   5.349546e-03 chr11  72968848
# cg04885068  0.19151186  0.1243965  0.25862722  6.474553 7.748434e-08 2.492609e-03   3.056447e-02  chr1   9099298
# cg05041382 -0.12420745 -0.1738795 -0.07453543 -6.391825 1.972869e-06 2.882289e-02   7.782181e-01 chr13  50366546
# cg05249026 -0.16819842 -0.2292563 -0.10714053 -2.355604 1.969621e-07 4.570217e-03   7.769369e-02 chr16  11031524
# cg06051716  0.35158015  0.2386962  0.46446411  6.493667 6.006034e-09 3.948567e-04   2.369140e-03  chr6 169225783
# cg06184647 -0.14036695 -0.1981679 -0.08256602 -5.918058 3.691896e-06 4.550954e-02   1.000000e+00 chr16  88450241
# cg06756211 -0.20482558 -0.2703873 -0.13926391 -6.049441 5.430893e-09 3.948567e-04   2.142270e-03  chr1 223936799
# cg06883949 -0.12332974 -0.1702170 -0.07644249 -6.468868 6.193470e-07 1.221538e-02   2.443076e-01  chr4   1795240
# cg08072480 -0.13914759 -0.1946040 -0.08369123 -6.612619 1.829518e-06 2.795438e-02   7.216717e-01 chr11  65292666
# cg10088075 -0.11123790 -0.1567888 -0.06568701 -6.213891 3.283303e-06 4.177845e-02   1.000000e+00 chr16  30407103
# cg11480534  0.39676891  0.2434952  0.55004268  5.838834 9.041808e-07 1.621196e-02   3.566632e-01 chr11  62255850
# cg11521780 -0.13675856 -0.1830387 -0.09047839 -5.767858 2.907204e-08 1.433469e-03   1.146776e-02 chr12  53693949
# cg11755803  0.40909980  0.2480944  0.57010521  5.790724 1.384129e-06 2.373842e-02   5.459837e-01 chr12 129000542
# cg12700904 -0.10557549 -0.1476654 -0.06348553 -5.936497 1.842554e-06 2.795438e-02   7.268139e-01  chr3  13590720
# cg12828656 -0.15460132 -0.2024569 -0.10674575 -0.497092 1.839642e-09 1.814163e-04   7.256650e-04  chr5 132577108
# cg13834844 -0.15424816 -0.2156167 -0.09287966 -5.125143 1.760982e-06 2.795438e-02   6.946369e-01 chr10 102415320
# cg13848566 -0.16585963 -0.2254945 -0.10622479 -6.364105 1.536174e-07 3.787246e-03   6.059593e-02  chr9  89561175
# cg14139125 -0.10741648 -0.1513767 -0.06345629 -5.828491 3.242799e-06 4.177845e-02   1.000000e+00  chr6 170893945
# cg14149680 -0.29164138 -0.3799804 -0.20330239 -6.352664 8.893297e-10 1.169350e-04   3.508050e-04 chr11   2187632
# cg16560774 -0.16783706 -0.2252284 -0.11044577 -6.459545 3.914132e-08 1.715521e-03   1.543969e-02 chr17  79373381
# cg19407717  0.49362683  0.3503547  0.63689896  4.033610 1.960244e-10 3.866189e-05   7.732378e-05  chr2   1544120
# cg19511862 -0.10431680 -0.1433824 -0.06525119 -6.258253 4.302669e-07 9.429059e-03   1.697231e-01 chr12 125478458
# cg19641404 -0.11345154 -0.1592483 -0.06765482 -5.865794 2.418284e-06 3.406844e-02   9.539163e-01  chr2 232826272
# cg20414082  0.25789189  0.1607362  0.35504761  3.200594 4.984959e-07 1.034930e-02   1.966367e-01  chr1 203002593
# cg20504533 -0.21837520 -0.2815800 -0.15517036 -5.852235 1.759757e-10 3.866189e-05   6.941537e-05 chr17  15848253
# cg22563312 -0.16772140 -0.2269815 -0.10846129 -6.274307 9.666561e-08 2.723623e-03   3.813072e-02  chr5 179334739
# cg22702772 -0.15069114 -0.2035174 -0.09786490 -6.685949 7.814700e-08 2.492609e-03   3.082587e-02  chr3  48699012
# cg24196240  0.41738155  0.2687882  0.56597491  5.267386 1.186723e-07 3.120764e-03   4.681146e-02 chr21  36096009
# cg24948962 -0.21864415 -0.2941613 -0.14312700 -5.437432 5.191151e-08 2.047701e-03   2.047701e-02 chr18   2906032
# cg25152909  0.43739500  0.2837949  0.59099512  6.076684 8.214755e-08 2.492609e-03   3.240392e-02  chr5 172566090
# cg25466588 -0.09339182 -0.1313223 -0.05546131 -5.901716 2.758046e-06 3.751513e-02   1.000000e+00  chr3  23958904

# comparison to cord blood DMPs
DMP_Mn_AdjM[DMP_Mn_AdjM$cpg %in% c('cg02042823', 'cg03763518'),]

                  # cpg      logFC       CI.L        CI.R   AveExpr         t      P.Value adj.P.Val         B adj.P.Val.bonf    std_err
# cg02042823 cg02042823  0.5004490  0.2462370  0.75466106  6.091279  3.888079 0.0001481948 0.3386672 0.6831844              1 0.12974758
# cg03763518 cg03763518 -0.1985491 -0.2987878 -0.09831043 -1.058518 -3.912046 0.0001354009 0.3276701 0.7573423              1 0.05095101

lambda(DMP_Mn_AdjM$P.Value)
# 0.9962195

# combp
DMP_Mn_AdjM = cbind(DMP_Mn_AdjM, anno[,'chr'][match(rownames(DMP_Mn_AdjM), rownames(anno))])
DMP_Mn_AdjM = cbind(DMP_Mn_AdjM, anno[,'pos'][match(rownames(DMP_Mn_AdjM), rownames(anno))])
DMP_Mn_AdjM$end = DMP_Mn_AdjM[,13]
DMP_Mn_AdjM_p = DMP_Mn_AdjM[,c(12,13,14,7,1)]
colnames(DMP_Mn_AdjM_p) = c( "chr","start", "end","p", "probe")
DMP_Mn_AdjM_p$chr = as.numeric(gsub("chr", "", DMP_Mn_AdjM_p$chr))

dir.create('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_M_age7')
setwd('/Users/annebozack/Documents/Cardenas/viva_DNAm_metals_local/combp_Mn_M_age7')
combp2(DMP_Mn_AdjM_p, region_plot = F, mht_plot = F, seed = 0.001, verbose = T)

# Number of DMRs identified:   12

# # chr	start	end	p	length	fdr	sidak	nprobe
# 1	161008461	161008977	5.55E-15	516	6.65E-14	4.24E-12	8
# 19	21657528	21658001	2.52E-14	473	1.51E-13	2.10E-11	5
# 19	10736005	10736117	1.47E-09	112	5.20E-09	5.19E-06	5
# 20	5485244	5485511	1.73E-09	267	5.20E-09	2.56E-06	6
# 4	79971176	79971432	3.59E-09	256	8.03E-09	5.53E-06	4
# 20	35169593	35169886	4.02E-09	293	8.03E-09	5.41E-06	5
# 1	9099280	9099462	6.96E-08	182	1.19E-07	0.000150813	3
# 10	45406680	45406847	1.28E-07	167	1.91E-07	0.000301199	3
# 3	39543966	39544192	2.93E-06	226	3.90E-06	0.005093047	2
# 15	91473090	91473365	7.73E-06	275	8.67E-06	0.011027659	4
# 19	12444612	12444741	7.95E-06	129	8.67E-06	0.024020202	5
# 5	70883064	70883087	2.07E-05	23	2.07E-05	0.29849227	3

rm(DMP_Mn_AdjM_p, DMP_Mn_AdjM);gc()

# EWAS using Beta values
# Adjusted for sex, race, GA, maternal age, maternal BMI, materal college grad, income >70k, nulliparous, smoking during pregnancy, and cell type
modAdjM = model.matrix(~ Mn_log2 + race_child2 + gestage_wks_deliv_d + age_mom_enroll_d + bmi_mom_prepreg_d + coll_grad + nullip + gt70k + smk_preg + Bcell_S_7y + CD4T_S_7y + CD8T_S_7y + Neu_S_7y + Mono_S_7y + NK_S_7y, data = pDatAge7MetalM)

# Beta-values for comoplete cases
ComBat.Betas.noMissAdjM = ComBat.Betas.Metals[,colnames(ComBat.Betas.Metals) %in% rownames(modAdjM)]
ComBat.Betas.noMissAdjM <- ComBat.Betas.noMissAdjM[,match(rownames(modAdjM), colnames(ComBat.Betas.noMissAdjM))]
all(rownames(modAdjM)==colnames(ComBat.Betas.noMissAdjM))
# TRUE 
identical(rownames(modAdjM),colnames(ComBat.Betas.noMissAdjM))
# TRUE 

DMP_Mn_Adj_betasM <- run_DMP(mvals = ComBat.Betas.noMissAdjM, design = modAdjM)

table(DMP_Mn_Adj_betasM$P.Value < 0.05)
# FALSE   TRUE 
# 375215  19245 

table(DMP_Mn_Adj_betasM$adj.P.Val < 0.05)
 # FALSE   TRUE 
# 394195    265 

DMP_Mn_Adj_betasM[DMP_Mn_Adj_betasM$cpg %in% c('cg02042823', 'cg03763518'),]


rm(DMP_Mn_Adj_betasM, ComBat.Betas.noMissAdjM);gc()
```
